web/src/main/java/org/wso2/testgrid/web/api/TestPlanService.java

```diff
@@ -107,4 +129,126 @@ public Response getTestPlan(@PathParam("id") String id, @QueryParam("require-tes
                     new ErrorResponse.ErrorResponseBuilder().setMessage(msg).build()).build();
         }
     }
+
+    /**
+     * Returns the log content related to the given test plan.
+     *
+     * @param id       test plan id to get the specific log
+     * @param truncate whether the log file should be truncated or not
+     * @return The requested Test-Plan
+     */
+    @GET
+    @Path("/log/{id}")
+    public Response getLogContent(@PathParam("id") String id, @QueryParam("truncate") boolean truncate) {
+        try {
+            // Get test plan
+            TestPlanUOW testPlanUOW = new TestPlanUOW();
+            Optional<TestPlan> optionalTestPlan = testPlanUOW.getTestPlanById(id);
+            if (!optionalTestPlan.isPresent()) {
+                String msg = "No test plan found for the given id " + id;
+                logger.error(msg);
+                return Response.serverError()
+                        .entity(new ErrorResponse.ErrorResponseBuilder().setMessage(msg).build()).build();
+            }
+            TestPlan testPlan = optionalTestPlan.get();
+
+            String testGridArtifactLocation = TestGridUtil.getTestRunArtifactsDirectory(testPlan).toString();
+            String key = Paths
+                    .get(AWS_BUCKET_ARTIFACT_DIR, testGridArtifactLocation, TESTGRID_LOG_FILE_NAME).toString();
+            ArtifactReadable artifactDownloadable = new AWSArtifactReader(AWS_REGION_NAME, AWS_BUCKET_NAME);
+
+            if (truncate) {
+                // The log file will be maximum of 50kb
+                ImmutablePair<Boolean, OutputStream> immutablePair = artifactDownloadable.readArtifact(key, 2);
```

Comment:
Comment says 50kb, but you are passing 2kb here.

And, 50kb is a bit low. Shall we go for 200kb? Lots of resources out in the wild of www has resources of that size.